### PR TITLE
[CIVP-10806] Add more utility functions for working with Civis files

### PR DESCRIPTION
**Pull request by stephen-hoover**
Add the functions `civis.io._files.file_id_from_run_output`, `civis.io._files.file_to_dataframe`, and `civis.io._files.file_to_json`. These functions will be used for the forthcoming modeling interface code, but are also of general interest.

### CODE REVIEW

**Comment by stephen-hoover** [comment]
@skomanduri , @keithing , would one of you be able to review this?

File: civis/tests/test_io.py [hunk 1]
```diff
@@ -200,3 +208,114 @@ def test_download_file(self, *mocks):
             with open(tmp.name, "r") as f:
                 data = f.read()
         assert data == expected
+
+
+def test_file_id_from_run_output_exact():
+    m_client = Mock()
+    m_client.scripts.list_containers_runs_outputs.return_value = \
```
Comment (keithing) [inline civis/tests/test_io.py]: Why not continue the line with 
```python
... = [
Response(...)
]
```
?

File: civis/io/_files.py [hunk 2]
```diff
@@ -167,3 +180,154 @@ def _get_url_from_file_id(file_id, client):
...
+    ------
+    ImportError
+        If ``pandas`` is not available
+
+    See Also
+    --------
+    :func:`~pandas.read_csv`
+    """
+    # Make sure we have pandas; this will raise an ImportError if not.
+    import pandas  # NOQA
```
Comment (keithing) [inline civis/io/_files.py]: For constancy with other places in the code we should set a global `HAS_PANDAS` in the `except ImportError` above and raise an error here like

```python
if not HAS_PANDAS:
    raise ImportError('file_to_dataframe requires pandas to be installed.')
```

File: civis/io/_files.py [hunk 3]
```diff
@@ -167,3 +180,154 @@ def _get_url_from_file_id(file_id, client):
...
+    file_url = file_info.file_url
+    file_name = file_info.name
+    if compression == 'infer':
+        for ext in ['.bz2', '.zip', '.xz']:
+            if os.path.splitext(file_name)[-1] == ext:
+                compression = ext[1:]
+                break
+        else:
+            if os.path.splitext(file_name)[-1] == '.gz':
+                compression = 'gzip'
```
Comment (keithing) [inline civis/io/_files.py]: Do we want compression to be `None` if nothing matches? Something like

```python
compression_exts = {'.gz': 'gzip', '.xz': 'xz', '.bz2': 'bz2', '.zip': 'zip'}
ext = os.path.splitext(file_name)[-1]
compression = compression_exts.get(ext)
```

Comment (stephen-hoover) [inline civis/io/_files.py]: I'd rather leave it as "infer" if we can't figure it out. In practice, I think we'll always end up with no compression in that case, but I believe pandas does check http headers to see if any compression format is indicated there.

Your code suggestion is still much clearer than what I wrote, so I changed to what you suggested (with an extra `if`).

File: civis/io/_files.py (hunk 2 above)
Comment (stephen-hoover) [inline civis/io/_files.py]: Done.

File: civis/tests/test_io.py (hunk 1 above)
Comment (stephen-hoover) [inline civis/tests/test_io.py]: Yes, that's better. Changed.

**Comment by keithing** [comment]
Awesome.  I imagine these functions will be useful.  LGTM!

**Pull request merged**